### ISSUE TITLE
Studies missing file_type in yaml file

### ISSUE THREAD

**Issue by jiyue1214** [EBISPOT/goci#1479]
@earlEBI found studies which missing the file_type in the yaml file. 

Class 1: there are two yaml files for each study, one with file_type and the other not. @karatugo, could you check if there file_type is available in the mangodb? I listed the files_type in another yaml file.

GCST90012642 pre-GWAS-SSF
GCST90012674 pre-GWAS-SSF
GCST90012783 pre-GWAS-SSF
GCST90012739 pre-GWAS-SSF

if they are in the mango, I can delete the other yaml.

Class 2: missing file types
@jiyue1214 need to extract headers and evaluate the file_type[other_empty-yamls-28.10.24.txt](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBMkNGQnc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--a411aaf3a42422275ad450576190fbac67943b75/other_empty-yamls-28.10.24.txt)

**Comment by jiyue1214** [comment]
Originally from the ticket (Empty yaml file types and incorrect formatting)[https://github.com/EBISPOT/goci/issues/1424]

**Comment by karatugo** [comment]
> @earlEBI found studies which missing the file_type in the yaml file.
> 
> Class 1: there are two yaml files for each study, one with file_type and the other not. @karatugo, could you check if there file_type is available in the mangodb? I listed the files_type in another yaml file.
> 
> GCST90012642 pre-GWAS-SSF GCST90012674 pre-GWAS-SSF GCST90012783 pre-GWAS-SSF GCST90012739 pre-GWAS-SSF
> 
> if they are in the mango, I can delete the other yaml.

All missing from mongodb.

**Comment by jiyue1214** [comment]
For studies, Lizzy mentioned missing the file_type in `other_empty-yamls-28.10.24.txt`, I extracted the headers
Here is the [result](https://docs.google.com/spreadsheets/d/1OTtXamLPqI_gGDgt9LFlkMxAoP8bYdFIl2_Sxz4ZB-c/edit?gid=0#gid=0):
These 84 studies can be classified into three groups
1. The file type is `GWAS-SSF v1.0`, which can directly be assigned.
2. The file type is `pre-GWAS-SSF`, which can directly be assigned.
3. 4 studies that the file type is `pre-GWAS-SSF`, also need to delete the "*-meta.yaml-meta.yaml" file after update

**Comment by jiyue1214** [comment]
Follow the previous [protocol](https://app.zenhub.com/workspaces/gwas-59df823c4a6feb3786810391/issues/gh/ebispot/goci/1441), create running.log for GCST004062 since it do not have other_allele.

**Comment by jiyue1214** [comment]
@karatugo Files types of studies mentioned by Lizzy are in the [result](https://docs.google.com/spreadsheets/d/1OTtXamLPqI_gGDgt9LFlkMxAoP8bYdFIl2_Sxz4ZB-c/edit?gid=0#gid=0) file. Could you please fill it in the database and regenerate yaml files for them. Thanks.

**Comment by karatugo** [comment]
will be done after the metadata yaml fix deployed

**Comment by karatugo** [comment]
File type for all but `GCST003371` and `GCST003372` are updated. These two accession IDs don't exist in studies table.

**Comment by karatugo** [comment]
Can't find their PMIDs either. To discuss in the meeting.

**Comment by jiyue1214** [comment]
GCST003371 and GCST003372 are old studies from 2016. It seems they do not have ingest API and only rest API. 

In addition, GCST003371 and GCST003372 follow the structure that: raw data not in GCST format, harmonised data named in the format of PMID-GCST-EFOid.h.tsv.gz. They should only contain the yaml file for harmonised data. what is described in the[ #1457](https://app.zenhub.com/workspaces/gwas-59df823c4a6feb3786810391/issues/gh/ebispot/goci/1457).

Currently, GCST003371 and GCST003372 already have yaml for *.h.tsv.gz.

**Comment by jiyue1214** [comment]
Most studies are done. Only one type of data that I am not sure about:

- raw sumstat: `pre-GWAS-SSF` because it has all mandatory columns but not in the specific order.
- harmonised sumstat:  Because we output them in the specific order and they satisfy the `GWAS-SSF V1.0` format

In this case, should we labelled the  harmonised sumstat as `GWAS-SSF V1.0` or `pre-GWAS-SSF`

**Comment by karatugo** [comment]
@karatugo Add file types to Mongo DB for GCST003371 and GCST003372

**Comment by karatugo** [comment]
GCST003371 and GCST003372 - Marked as pending again.

**Comment by karatugo** [comment]
After the [fix](https://app.zenhub.com/workspaces/gwas-59df823c4a6feb3786810391/issues/gh/ebispot/gwas-sumstats-service/374) is deployed, yamls will be generated correctly.

**Comment by karatugo** [comment]
The fix for this issue is deployed to prod. Expect new yamls in a few days.

**Comment by ljwh2** [comment]
Waiting for ftp sync to complete

**Comment by karatugo** [comment]
yamls are generated but they are not synced to public ftp. This is because file name does not start with `GCST` (ftp sync only syncs files that start with `GCST`).

**Comment by karatugo** [comment]
There are a few options to solve this problem. Let me know which one sounds best. I'm leaning towards Option 1 as there are not many freely named files. We can think of them as exceptions.  

1. Copy files manually
2. Rename files with GCST IDs and let ftp sync handle it automatically
3. Update sync logic

**Comment by ljwh2** [comment]
I agree, handle these manually. We don't want to rename them with GSCT as this marks them out as being shared prior to any submission format.

**Comment by karatugo** [comment]
copied them manually 

https://ftp.ebi.ac.uk/pub/databases/gwas/summary_statistics/GCST003001-GCST004000/GCST003371/
https://ftp.ebi.ac.uk/pub/databases/gwas/summary_statistics/GCST003001-GCST004000/GCST003372/
